src/webapp/components/ClusterMapLayers.jsx

```diff
@@ -3,58 +3,77 @@
    If the state belongs to the same component as the Map/BasicMap, it re-renders
    the map on any state change, causing the viewport and zoom to be reset
    */
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useContext } from "react";
 import { Layer, MapContext } from "react-mapbox-gl";
 import RequestPopup from "./RequestPopup";
+import ClusterMapContext from "../context/ClusterMapContext";
 
 const handleClusterOnClick = (map, e, layerId, sourceId) => {
   const features = map.queryRenderedFeatures(e.point, {
-    layers: [layerId]
+    layers: [layerId],
   });
   const clusterId = features[0].properties.cluster_id;
+
   map.getSource(sourceId).getClusterExpansionZoom(clusterId, (err, zoom) => {
     if (err) return;
 
     map.easeTo({
       center: features[0].geometry.coordinates,
-      zoom
+      zoom,
     });
   });
 };
 
 const makePopupData = (features, lngLat) => {
-  return features.map(feat => {
+  return features.map((feat) => {
     const metaSrc = feat.properties.meta;
     const meta =
       typeof metaSrc == "string" ? JSON.parse(feat.properties.meta) : metaSrc;
     return {
       lngLat,
-      meta
+      meta,
     };
   });
 };
 
-const ClusterMapLayers = ({ geoJsonData, paramRequest, sourceId, color }) => {
+const ClusterMapLayers = ({ data, paramRequest, sourceId, color }) => {
   const [popup, setPopup] = useState();
+  const [initialRender, setInitialRender] = useState(true);
+  const { focusedRequestId, setFocusedRequestId } = useContext(
+    ClusterMapContext
+  );
+
+  const setCursorPointer = (e) =>
+    (e.target.getCanvas().style.cursor = "pointer");
+  const setCursorGrab = (e) => (e.target.getCanvas().style.cursor = "grab");
 
-  const setCursorPointer = e => e.target.getCanvas().style.cursor = 'pointer';
-  const setCursorGrab = e => e.target.getCanvas().style.cursor = 'grab';
 
-  // display popup if request code is present in URL search param
   useEffect(() => {
-    if (paramRequest) {
-      const {
-        geometry: { coordinates }
-      } = paramRequest;
-      setPopup(makePopupData([paramRequest], coordinates));
+    // display popup if request code is present in URL search param
+    // but only on first render
+    if (paramRequest && initialRender) {
+      setPopup(makePopupData([paramRequest], paramRequest.geometry.coordinates));
+      setInitialRender(false)
+    }
+
+    //find focused request and display popup
+    const [focusedRequest] = data.filter(
+        ({
+          properties: {
+            meta: { Code },
+          },
+        }) => Code === focusedRequestId
+      );
+    if(focusedRequest){
+      setPopup(makePopupData([focusedRequest], focusedRequest.geometry.coordinates))
     }
-  }, []);
+  }, [focusedRequestId]);
```

Comment:
not related to this PR, but surprised we don't need paramRequest here as well. wouldn't an in-app navigation with the code param reuse the same paramRequest otherwise?

Reply:
thats a good question. so unfortunately, the way i set up paramRequest before was that it only worked on the first map render, so updating the url param wouldn't re-focus the request. though your comm ent made me think that we should update the url param on focus. it would be easier for me to remove the old paramRequest logic and replace handling url params with `focusedRequest` instead.